library/Fission/Web/Auth.hs

```diff
@@ -1,30 +1,30 @@
 module Fission.Web.Auth
-  ( server
-  , context
+  ( Checks
+  , server
   , basic
   , user
   , checkUser
   ) where
 
 import Crypto.BCrypt
-import Database.Selda
+import Database.Esqueleto
 import Servant
 
+import Fission.Models
 import Fission.Prelude
-import Fission.Storage.Query
-import Fission.User as User hiding (username)
-import Fission.Web.Server
+
+type Checks = '[BasicAuthCheck (Entity User), BasicAuthCheck ByteString]
 
 server
-  :: HasServer api '[BasicAuthCheck User, BasicAuthCheck ByteString]
+  :: HasServer api Checks
   => Proxy api
-  -> cfg
-  -> RIOServer cfg api
-  -> Server api
-server api cfg = hoistServerWithContext api context (toHandler cfg)
-
-context :: Proxy (BasicAuthCheck User ': BasicAuthCheck ByteString ': '[])
-context = Proxy
+  -> (forall a . m a -> Handler a)
```

Comment:
Same note?

We actually do need this type to be existential:

```haskell
    • Couldn't match type ‘m a -> Handler a’
                     with ‘forall x. m0 x -> n0 x’
      Expected type: (m a -> Handler a)
                     -> ServerT api m -> ServerT api Handler
        Actual type: (forall x. m0 x -> n0 x)
                     -> ServerT api m0 -> ServerT api n0
```

Reply:
Yeah, I had a similar comment here: https://github.com/fission-suite/web-api/pull/226#discussion_r358257850

I wondered if it might be useful to note down, in a code comment, why we have the `forall`.

Reply:
Sure, I'll throw a comment in here 👍  It's a standard part of the type system, but it's not in a lot of place in the code, so makes sense.